src/indicator/nmofono/manager-impl.cpp

```diff
@@ -479,6 +479,24 @@ public Q_SLOTS:
         }
 
         Q_EMIT p.simForMobileDataChanged();
+
+        if (sim != nullptr){
+            QDBusConnection bus = QDBusConnection::systemBus();
+            if (!bus.isConnected())
+            {
+                assert(0);
+            }
+            QDBusInterface dbus_iface(OFONO_SERVICE,
+                                      OFONO_MANAGER_PATH,
+                                      "org.nemomobile.ofono.ModemManager",
+                                      bus);
+
+            if ( dbus_iface.isValid() ) {
+                QDBusPendingReply<QVariantMap> reply = dbus_iface.asyncCall("SetDefaultDataSim",
+                                                                m_simForMobileData->imsi());
+                reply.waitForFinished();
```

Comment:
There's no point in making an async call if you are waiting for its result anyway.

Reply:
true, I changed this